sdk/monitor/monitor-opentelemetry/src/shared/logging/diagFileConsoleLogger.ts

```diff
@@ -115,6 +119,48 @@ export class DiagFileConsoleLogger implements DiagLogger {
     }
   }
 
+  /**
+   * Checks if the warning message is related to accessing resource attributes
+   * before async attributes are settled and should be filtered out
+   */
+  private _shouldFilterResourceAttributeWarning(message?: any, args?: any[]): boolean {
+    const messagesToFilter = [
+      "Accessing resource attributes before async attributes settled",
+      "Resource attributes being accessed before async attributes finished",
+      "async attributes settled",
+      "Resource attributes accessed before async detection completed",
+    ];
+
+    if (typeof message === "string") {
+      // Filter out warnings about accessing resource attributes before async attributes are settled
+      return messagesToFilter.some((filterText) => message.includes(filterText));
+    }
+
+    // Check if the message is in the args array
+    if (args && Array.isArray(args)) {
+      for (const arg of args) {
+        if (typeof arg === "string") {
+          if (messagesToFilter.some((filterText) => arg.includes(filterText))) {
+            return true;
+          }
+        }
+      }
+    }
+
+    // Also check if message starts with the warning text (in case it's formatted differently)
+    if (typeof message === "string") {
+      const messageParts = message.split(" ");
+      if (messageParts.length >= 3) {
+        const messageStart = `${messageParts[0]} ${messageParts[1]} ${messageParts[2]}`;
```

Comment:
Why construct a new string and not just check that
- There are 3 components
- Check each component for an exact match (using if (x && y &&..) as this would also shortcut the comparison of subsequent "parts" when the first one does not match

Reply:
Updated this to use the logic suggested here.